app/src/main/java/com/asma/tasky/feature_authentication/presentation/components/FormTextField.kt

```diff
@@ -0,0 +1,149 @@
+package com.asma.tasky.feature_authentication.presentation.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.asma.tasky.R
+import com.asma.tasky.core.presentation.ui.theme.Green
+import com.asma.tasky.core.presentation.ui.theme.LightGray
+
+
+@Composable
+fun FormTextField(
+    modifier: Modifier = Modifier,
+    text: String = "",
+    hint: String = "",
+    maxLength: Int = 400,
+    error: String = "",
```

Comment:
Here I'd again prefer a nullable field over an empty string. An empty string for an error field to me means there is an error, but no message 🤔